sdks/go/pkg/beam/core/runtime/exec/sideinput.go

```diff
@@ -58,7 +59,16 @@ func NewSideInputAdapter(sid StreamID, sideInputID string, c *coder.Coder, wm Wi
 }
 
 func (s *sideInputAdapter) NewIterable(ctx context.Context, reader StateReader, w typex.Window) (ReStream, error) {
-	key, err := EncodeElement(s.kc, []byte(iterableSideInputKey))
+	return s.NewKeyedIterable(ctx, reader, w, iterableSideInputKey)
+}
+
+func (s *sideInputAdapter) NewKeyedIterable(ctx context.Context, reader StateReader, w typex.Window, iterKey interface{}) (ReStream, error) {
+	// TODO(jrmccluskey): re-write to ensure support for non-string keys
```

Comment:
I don't love this little hack, it will need a more substantial re-write to support non-string keys. For now this establishes the function signature to be correct; however, it will have to get tweaked once I get the end-to-end code change completed.

Reply:
What failure do you get without this? AFAICT, it should be plumbing the key coder for the SideInput's PCollection properly...

Reply:
Also, use TODO(BEAM-3293) instead of your username.

Reply:
Realized the fixed key is bytes encoded, not a string. That's why it was `[]byte(iterableSideInputKey)` before. Fixed the argument passed in NewIterable() to arrange this beforehand. Cleaned it up and removed the TODO